targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/serial_api.c

```diff
@@ -1082,6 +1089,19 @@ void serial_init(serial_t *obj, PinName tx, PinName rx)
         uart_object->rx = rx;
     }
 
+		/*
+		 * For the console UART (instance 0), it is seen that the Easy DMA based
+		 * RX does not fill the FIFO as expected when the Flow Control is uninitialized.
+		 * Even though we set the flow control once here, it gets reset subsequentially.
+		 * This seems necessary until a better fix is found as users can specify RawSerial
+		 * interface with no flow control, resulting in broken RX.
+		 */
```

Comment:
So by default  UART0 will use flow control connected to the terminal:
- Flow control enabled by default
- Flow control can be explicitly enabled or disabled by user by calling `Serial::set_flow_control`

If this is the case it seems reasonable to me.